### PR TITLE
Make inline select searchable for table

### PR DESCRIPTION
**Pull request by Marswang92**
# Change Log
- Add a `searchable` props for inline select.
- Use the search bar to filter dropdown list according to its value.

gaz6rkmy8b

### CODE REVIEW

File: table/sandbox/sandbox.jsx [hunk 1]
```diff
@@ -33,6 +33,7 @@ const renderCell = (rowData, columnDef) => {
         { key: 11, value: 'Burmese Bamboo Shark' },
         { key: 12, value: 'Megamouth Shark' }
       ]}
+      searchable={true}
```
Comment (ben-burlingham) [inline table/sandbox/sandbox.jsx]: nb No need for `true`, it's implicit

File: table/src/InlineSelect.scss [hunk 2]
```diff
@@ -97,6 +97,14 @@
   z-index: 1;
 }
 
+.searchInput {
+  width: calc(100% - 10px);
```
Comment (ben-burlingham) [inline table/src/InlineSelect.scss]: Avoid `calc` at all times, bad perf and almost always unnecessary

File: table/src/InlineSelect.scss [hunk 3]
```diff
@@ -97,6 +97,14 @@
...
+  padding: 3px;
+  margin: 5px;
+  border-radius: 2px;
+  border: 1px solid $gray;
+}
```
Comment (ben-burlingham) [inline table/src/InlineSelect.scss]: Please alphabetize CSS props

File: table/src/InlineSelect.jsx [hunk 4]
```diff
@@ -116,26 +109,55 @@ export default class InlineSelect extends React.Component {
...
-            {item.value}
-          </div>)
-        )
-      }
-    </div>
-  )
+  showDropdown = (e) => {
+    if (!e || e.target !== this.clear) {
+      this.setState({ expanded: true }, this.position);
+      document.addEventListener('click', this.hideDropdown);
```
Comment (ben-burlingham) [inline table/src/InlineSelect.jsx]: This, and the listener below, are never removed.  This particle uses the `tabindex`/focus/blur pattern, not event listeners.  You'll have to use the existing `onMouseDown` pattern, which implicitly manages event listeners and performs well.  See `InlineDateSelect` for an example, or swing by my desk and I'll show you how.

Comment (Marswang92) [inline table/src/InlineSelect.jsx]: It is removed when hideDropdown is triggered, so it will automatically clean itself.

**Pull request merged**